IdentityServer/v7/docs/content/fundamentals/license_key.md

```diff
@@ -1,42 +1,115 @@
+including environment variables, appsettings.json, an external configuration service such
+as Azure App Configuration, etc.
 
-The contents of the license key file is text, and so that is the value to assign to the *LicenseKey* property.
-For example:
+The *AddIdentityServer* method accepts a lambda expression to configure various options in
+your IdentityServer, including the *LicenseKey*. Set the value of this property to the
+content of the license key file.
 
 ```csharp
 public void ConfigureServices(IServiceCollection services)
 {
     services.AddIdentityServer(options =>
     {
-        options.LicenseKey = "eyJhbG..."; // the contents of the license key file
+        options.LicenseKey = "eyJhbG..."; // the content of the license key file
     });
 }
 ```
 
 ## License Validation and Logging
 
-All license validation is self-contained and does not leave the host (meaning there are no outbound calls related to license validation).
-Any messages from the license validation layer will be emitted to the logging system.
-The level of the log entry depends on the nature of the message and the type of license.
+The license is validated at startup and during runtime. All license validation is
+self-contained and does not leave the host. There are no outbound calls related to license
+validation.
+
+#### Startup Validation
+At startup, IdentityServer first checks for a license. If there is no license configured,
+IdentityServer logs a warning indicating that a license is required in a production
+deployment. You can ignore these messages in non-production environments.
+
+Next, assuming a license is configured, IdentityServer compares its configuration to the
+license. If there are discrepancies between the license and the configuration,
+IdentityServer will write log messages indicating the nature of the problem.
+
+
+#### Runtime Validation
+Most common licensing issues, such as expiration of the license or configuring more
+clients than is included in the license do not prevent IdentityServer from functioning. We
+trust our customers and we don't want a simple oversight to cause an outage. However, some
+features will be disabled at runtime if your license does not include them, including:
+
+- Server Side Sessions
+- DPoP
+- Resource Isolation
+- PAR
+- Dynamic Identity Providers 
+- CIBA
+
+Again, the absence of a license is permitted for development and testing, and therefore
+does not disable any of these features.
+
+## Redistribution
+We understand that when IdentityServer is redistributed, log messages from the licensing
+system are not likely to be very useful to your redistribution customers. For that reason,
+in a redistribution the severity of log messages from the license system is turned all the
+way down to the Trace level. We also appreciate that it might be cumbersome to deploy
+updated licenses in this scenario. You are not required to deploy updated licenses to your
```

Comment:
I'd rather not imply that we think it's ok to not update the key to the latest if they can. I'd rephrase this along the lines of "we ask that you update the license key at the next deployment of your software" or somesuch.